### PR TITLE
Add switch OS, switch search queries and better datastore tests.

### PR DESCRIPTION
**Pull request by Gerrit91**
- takes out unrelated parts of adding switch OS from #351 
- adds additional search queries for switches allowing more efficient filtering in `metalctl`
- much better testing in the datastore layer for switches using a real rethinkdb as a backend

### CODE REVIEW

File: cmd/metal-api/main.go [hunk 1]
```diff
@@ -938,7 +938,7 @@ func run() error {
 	// if we will add support for api-tokens as headers, we had to add them
 	// here to. note: the token's should not contain the product (aka. metal)
 	// because customers should have ONE token for many products.
-	// ExposeHeaders:  []string{"X-FITS-TOKEN"},
+	// ExposeHeaders:  []string{"X-TOKEN"},
```
Comment (majst01) [inline cmd/metal-api/main.go:941]: unrelated

File: go.mod [hunk 2]
```diff
@@ -190,3 +190,5 @@ require (
 	nhooyr.io/websocket v1.8.7 // indirect
 	tailscale.com v1.32.3 // indirect
 )
+
+require gopkg.in/inf.v0 v0.9.1 // indirect
```
Comment (majst01) [inline go.mod:194]: wrong block

**Review by majst01 — COMMENTED** [review]
Like it, but then we must first test this with released metal-hammer and metal-core

File: cmd/metal-api/internal/datastore/rethinkdb_integration_test.go [hunk 3]
```diff
@@ -5,43 +5,61 @@ package datastore
...
-	"go.uber.org/zap/zaptest"
+	"github.com/testcontainers/testcontainers-go"
+	"go.uber.org/zap"
 
 	"testing"
-
-	"github.com/stretchr/testify/require"
 )
 
-func TestRethinkStore_ConflictIsReturned(t *testing.T) {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/datastore/rethinkdb_integration_test.go:18]: This is now already covered in a switch test and will also be covered in all other tests if we start to test like this for the rest of the package. Hence removed.

File: cmd/metal-api/internal/datastore/rethinkdb_test.go [hunk 4]
```diff
@@ -85,31 +85,6 @@ func TestRethinkStore_db(t *testing.T) {
 	}
 }
 
-func TestRethinkStore_Mock(t *testing.T) {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/datastore/rethinkdb_test.go:88]: This test is rather useless and also the linter does not like it, therefore removed.

**Comment by Gerrit91** [comment]
Works on the mini-lab (switch registration plus machine provisioning), so it's compatible with the current metal-core and metal-hammer. No breaking API changes included.

**Review by majst01 — APPROVED** [review]
